qemu/tests/virtio_fs_multi_users_access.py

```diff
@@ -0,0 +1,212 @@
+            win_driver_installer_test.run_viofs_service(test, params, session)
+            vm.reboot(session)
+        else:
+            error_context.context("Create a destination directory %s "
+                                  "inside guest." % fs_dest, test.log.info)
+            if not utils_misc.make_dirs(fs_dest, session=session):
+                test.fail("Creating directory was failed!")
+            error_context.context("Mount virtiofs target %s to %s inside"
+                                  " guest." % (fs_target, fs_dest),
+                                  test.log.info)
+            if not utils_disk.mount(fs_target, fs_dest, 'virtiofs',
+                                    session=session):
+                test.fail('Mount virtiofs target failed.')
+            error_context.context("Set 777 permission for all users.",
+                                  test.log.info)
+            session.cmd("chmod -R 777 %s" % fs_dest)
+
+        for _username, _pwd in zip(username, pwd):
+            try:
+                if not domain_dns:
+                    error_context.context("Login the user: %s" % _username,
+                                          test.log.info)
+                    session = vm.wait_for_login(username=_username,
+                                                password=_pwd)
+                else:
+                    session = vm.wait_for_login()
+                    eth_name = get_eth_name(session)
+                    switch_dynamic_ip_to_static_ip(session, eth_name)
+                    join_domain_cmd = params.get("join_domain")
+                    join_domain_cmd = join_domain_cmd.replace("%s",
+                                                              _username,
+                                                              1)
+                    join_domain_cmd = join_domain_cmd.replace("%s",
+                                                              _pwd,
+                                                              1)
+                    error_context.context("Join domain...", test.log.info)
+                    output = session.cmd_output(join_domain_cmd)
+                    if "does not exist" in output:
+                        test.fail("Failed to join the domain!")
+                    elif "is not recognized as an internal" in output:
+                        error_context.context("The netdom is NOT supported, "
+                                              "trying to use powershell...",
+                                              test.log.info)
+                        ps_cred = params.get("ps_cred")
+                        ps_join_domain = params.get("ps_join_domain")
+                        ps_cred = ps_cred % (_username, _pwd)
+                        ps_join_domain = ps_cred + ps_join_domain
+                        session.cmd("powershell \"" + ps_join_domain + "\"")
+                    vm.reboot(session)
```

Comment:
Be careful with reboot here, the original session died after the reboot, but line171 uses the session again.
I think your code is OK because line171 is for linux only , while reboot here is for windows only.  But one day if we want to add domain_dns for linux, this reboot can lead to error and it's hard to find it out. I remembered that we hit this kind of issue before. So I suggest to reset the session with the returned, e.g. session = vm.reboot(session).

Also please check other reboots, I think they are safe, but you can make a double check.

Reply:
> Be careful with reboot here, the original session died after the reboot, but line171 uses the session again. I think your code is OK because line171 is for linux only , while reboot here is for windows only. But one day if we want to add domain_dns for linux, this reboot can lead to error and it's hard to find it out. I remembered that we hit this kind of issue before. So I suggest to reset the session with the returned, e.g. session = vm.reboot(session).
> 
> Also please check other reboots, I think they are safe, but you can make a double check.

@zhencliu thanks for response, I'm on curiousness if we have domain_dns on linux.
Hi @xiagao, do we have domain_dns on linux scene for multi users accessing the shared directory? Thanks!

Reply:
hi @nickzhq , actually, my point is, vm.reboot returns a new session, if we still use a session after reboot, then we can use this one, i.e. no need to get a new one with wait_for_login function; if we never use a session after reboot, it's OK not to get the returned session, but a better way IMO is that we still get it from reboot, and do session.close() eventually.

Your code is OK for me, I left the note just like a reminder, because we hit 'reboot' issue before

Reply:
> > Be careful with reboot here, the original session died after the reboot, but line171 uses the session again. I think your code is OK because line171 is for linux only , while reboot here is for windows only. But one day if we want to add domain_dns for linux, this reboot can lead to error and it's hard to find it out. I remembered that we hit this kind of issue before. So I suggest to reset the session with the returned, e.g. session = vm.reboot(session).
> > Also please check other reboots, I think they are safe, but you can make a double check.
> 
> @zhencliu thanks for response, I'm on curiousness if we have domain_dns on linux. Hi @xiagao, do we have domain_dns on linux scene for multi users accessing the shared directory? Thanks!
For linux, there is no.

Reply:
> but a better way IMO is that we

Hello @zhencliu , I got your concern which 'reboot' issue was blocked to us previously. For now in this patch, I have paid the much more attentions on the 'reboot', and after I did the many times tests, those codes work well without 'reboot' issue. 
According to the response from owner, we do NOT need to pay attention on domain_dns linux.

Reply:
Hello @nickzhq , for @zhencliu 's reboot session opinion, I also think it's better to have the new session after vm reboot. The reason is to avoid the issues we hit before and if we want to extend the code, no session return would lead to some error, and we might need more time to debug.
BTW, the test on win11 passed.

Reply:
> Hello @nickzhq , for @zhencliu 's reboot session opinion, I also think it's better to have the new session after vm reboot. The reason is to avoid the issues we hit before and if we want to extend the code, no session return would lead to some error, and we might need more time to debug. BTW, the test on win11 passed.

Hello @xiagao , my opinion is that the `session = vm.reboot(session)` is NOT necessary at here unless we have a case like domain_dns on linux. However, if you guys really want to put `session = vm.reboot(session)`  in code, it's okay for me.